modules/user-access/pages/rbac-row-policy/rbac-row-policy.adoc

```diff
@@ -0,0 +1,513 @@
+[console]
+----
+GSQL > show function lib1.*
+create function lib1.func1() returns (bool) {
+RETURN true;
+}
+create function lib1.func2() returns (bool) {
+RETURN true;
+}
+----
+
+.Ex. Use `-r` to show all functions that start with `lib`:
+[console]
+----
+GSQL > show function -r "lib.*"
+create function lib1.func1() returns (bool) {
+RETURN true;
+}
+create function lib1.func2() returns (bool) {
+RETURN true;
+}
+create function lib2.func1() returns (bool) {
+RETURN true;
+}
+----
+
+.Ex. To show all functions available:
+[console]
+----
+GSQL > show function *
+create function lib1.func1() returns (bool) {
+RETURN true;
+}
+create function lib1.func2() returns (bool) {
+RETURN true;
+}
+create function lib2.func1() returns (bool) {
+RETURN true;
+}
+----
+
+== Object-Based Privileges
+
+This mechanism allows users to grant or revoke privileges based on specific objects.
+Allowing users to specify the privilege names, objects, and other details.
+
+=== Privilege Details
+
+Here are some important terms and details for object-based privileges:
+
+* *Privilege Names*: Are the specific privileges that can be granted or revoked, such as `ACCESS`, `CREATE`, `READ`, etc.
+* *Privilege Objects*: Are the objects to which the privileges apply.
+They can be things like `GLOBAL`, `VERTEX`, `EDGE`, etc.
+* *Privilege Scopes*: Define where these privileges apply, like `GRAPH`, `PACKAGE`, or `GLOBAL`.
+
+To see a complete list, as well as the legacy privilege syntax that the object-base privilege relate to, go to the xref:rbac-row-policy/row-policy-privlages-table.adoc[].
+
+=== Privilege Commands
+
+To grant or revoke object-based privileges, use these commands.
+
+.Ex. Granting Privileges:
+[console, gsql]
+----
+GRANT privilegeNames ON privilegeObjects IN privilegeScopes TO roleNames
+----
+
+.Ex. Revoking Privileges:
+[console, gsql]
+----
+REVOKE privilegeNames ON privilegeObjects IN privilegeScopes FROM roleNames
+----
+
+For users that grant privileges in the legacy privilege syntax, transitioning to object-based privilege syntax is simple.
+
+.Compare Legacy and Object based syntax:
+[cols="2", separator=¦ ]
+|===
+¦ Privilege Syntax ¦  Example
+
+¦  Legacy ¦ grant privilege READ_DATA, CREATE_DATA, UPDATE_DATA, DELETE_DATA on global to role1
+
+¦  Object-Based ¦ grant READ, CREATE, UPDATE, DELETE on ALL DATA in global to role1
+
+|===
+
+=== Privilege Protection
+Additionally, there are new RBAC privileges to protect the `FUNCTION` and `POLICY`.
+
+.FUNCTION
+[cols="3", separator=¦ ]
+|===
+¦ Privilege ¦ Levels ¦ Operations
+
+¦ USE ¦ global/package/function ¦
+
+show function
+
+apply row policy
+¦ WRITE ¦ global/package/function ¦ create/replace/drop function
+
+|===
+
+.Policy
+[cols="3", separator=¦ ]
+|===
+¦ Privilege ¦ Levels ¦ Operations
+
+¦ READ ¦ global/graph ¦ show applied policy
+¦ WRITE ¦ global/graph ¦ apply row policy
+
+clear row policy
+|===
+
+.For built-in roles, there are some changes as well:
```

Comment:
Need to add this to the built-in role privilege list as well.
https://docs.tigergraph.com/tigergraph-server/current/user-access/access-control-model#_built_in_roles

Reply:
Thank you. Added these to the list and added some links as well. Will push with the next commit.